// @progress/kendo-barcodes-vue-wrapper/dist/es/KendoQRCode/KendoQRCodeMixin.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _kendoBaseComponentsVueWrapper = require('@progress/kendo-base-components-vue-wrapper');

var _QRCodeProps = require('./QRCodeProps');

var _QRCodeProps2 = _interopRequireDefault(_QRCodeProps);

var _Border = require('./Border');

var _Border2 = _interopRequireDefault(_Border);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
    props: _QRCodeProps2.default,
    emits: {
        'kendowidgetready': null
    },
    mixins: [_kendoBaseComponentsVueWrapper.KendoBaseComponent, _Border2.default],
    mounted: function mounted() {
        kendo.jQuery(this.$el).kendoQRCode(this.widgetOptions);

        if (this.v3) {
            this.ready();
            this.$emit('kendowidgetready', this.kendoWidget());
        } else {
            this.$emit('kendowidgetready', this.kendoWidget());
        }
    },

    methods: {
        kendoWidget: function kendoWidget() {
            return kendo.jQuery(this.$el).getKendoQRCode();
        }
    }
};
