// @progress/kendo-vue-inputs/dist/npm/checkbox/Checkbox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CheckboxVue3 = exports.Checkbox = void 0; // @ts-ignore

var Vue = require("vue");

var allVue = Vue;
var gh = allVue.h;
var ref = allVue.ref;
var inject = allVue.inject;

var kendo_vue_common_1 = require("@progress/kendo-vue-common");

var package_metadata_1 = require("../package-metadata");

var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");

var messages_1 = require("./../messages");

var Checkbox = {
  name: 'KendoCheckbox',
  // @ts-ignore
  emits: {
    'changemodel': null,
    'update:modelValue': null,
    change: null,
    focus: null,
    blur: null
  },
  model: {
    event: 'changemodel'
  },
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  props: {
    checked: {
      type: Boolean,
      default: undefined
    },
    defaultChecked: {
      type: Boolean,
      default: undefined
    },
    defaultValue: {
      type: [String, Boolean],
      default: undefined
    },
    modelValue: {
      type: [String, Boolean],
      default: undefined
    },
    dir: String,
    disabled: Boolean,
    id: String,
    ariaLabelledBy: String,
    ariaDescribedBy: String,
    label: String,
    labelRender: [String, Number, Boolean, Object],
    labelPlacement: String,
    labelOptional: Boolean,
    name: String,
    tabIndex: Number,
    value: {
      type: [String, Boolean],
      default: undefined
    },
    validationMessage: String,
    required: Boolean,
    valid: {
      type: Boolean,
      default: undefined
    },
    validityStyles: {
      type: Boolean,
      default: true
    }
  },
  data: function data() {
    return {
      valueDuringOnChange: undefined,
      currentDir: 'ltr',
      currentChecked: undefined,
      currentValue: undefined
    };
  },
  created: function created() {
    kendo_vue_common_1.validatePackage(package_metadata_1.packageMetadata);
    this.calculatedId = kendo_vue_common_1.guid();

    if (this.$props.defaultChecked !== undefined) {
      this.currentChecked = this.$props.defaultChecked;
    }

    if (this.$props.defaultValue !== undefined) {
      this.currentValue = this.$props.defaultValue;
    }

    this.currentDir = this.$props.dir;
  },
  computed: {
    valueIsBooleanOrNull: function valueIsBooleanOrNull() {
      var value = this.$props.value;
      return typeof value === 'boolean' || value === null;
    },
    isCheckedControlled: function isCheckedControlled() {
      return this.$props.checked !== undefined;
    },
    isValueControlled: function isValueControlled() {
      return this.$props.value !== undefined && this.valueIsBooleanOrNull;
    },
    computedValue: function computedValue() {
      return this.$data.valueDuringOnChange !== undefined ? this.$data.valueDuringOnChange : this.$props.value !== undefined ? this.$props.value : this.$props.modelValue !== undefined ? this.$props.modelValue : this.$data.currentValue;
    },
    computedChecked: function computedChecked() {
      return this.$data.valueDuringOnChange !== undefined ? this.$data.valueDuringOnChange : this.$props.checked !== undefined ? this.$props.checked : this.$props.modelValue !== undefined ? this.$props.modelValue : this.$data.currentChecked;
    },
    useValueAsChecked: function useValueAsChecked() {
      return this.computedChecked === undefined && this.computedValue;
    },
    checkedProp: function checkedProp() {
      return this.useValueAsChecked ? this.computedValue : this.computedChecked;
    },
    valueProp: function valueProp() {
      var value = this.$props.value;
      return this.useValueAsChecked || this.isValueControlled ? value === null ? value : undefined : value || this.computedValue;
    },
    indeterminateProp: function indeterminateProp() {
      return this.checkedProp === null || this.valueProp === null;
    },
    isValid: function isValid() {
      var valid = this.$props.valid;
      return valid !== undefined ? valid : !this.$props.required ? true : this.computedChecked ? true : false;
    }
  },
  mounted: function mounted() {
    this.input = this.v3 ? this.inputRef : this.$refs.input;

    if (!this.currentDir && window && this.$el) {
      // Note: getComputedStyle forces reflow
      var direction = window.getComputedStyle(this.$el).direction;

      if (direction) {
        this.currentDir = direction;
      }
    }

    this.setValidity();
  },
  updated: function updated() {
    if (!this.input) {
      this.input = this.v3 ? this.inputRef : this.$refs.input;
    }

    this.setValidity();
  },
  // @ts-ignore
  setup: !gh ? undefined : function () {
    var v3 = !!gh;
    var inputRef = ref(null);
    var kendoLocalizationService = inject('kendoLocalizationService', {});
    return {
      v3: v3,
      inputRef: inputRef,
      kendoLocalizationService: kendoLocalizationService
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    var _a = this.$props,
        ariaDescribedBy = _a.ariaDescribedBy,
        ariaLabelledBy = _a.ariaLabelledBy,
        disabled = _a.disabled,
        id = _a.id,
        label = _a.label,
        labelRender = _a.labelRender,
        labelPlacement = _a.labelPlacement,
        name = _a.name,
        labelOptional = _a.labelOptional,
        tabIndex = _a.tabIndex,
        required = _a.required,
        validityStyles = _a.validityStyles;
    var defaultSlot = kendo_vue_common_1.getDefaultSlots(this);
    var renderedLabel = label;
    this.localizationService = kendo_vue_intl_1.provideLocalizationService(this);
    this.defaultValidationMessage = this.localizeMessage(messages_1.checkboxValidation);
    this.optionalMessage = this.localizeMessage(messages_1.checkboxOptionalText);
    var checkboxClasses = kendo_vue_common_1.classNames({
      'k-state-disabled': disabled
    });
    var inputClasses = kendo_vue_common_1.classNames({
      'k-checkbox': true,
      'k-state-indeterminate': this.indeterminateProp,
      'k-state-invalid k-invalid': !(this.isValid || validityStyles !== undefined || validityStyles === true)
    });

    var checkboxInput = function checkboxInput() {
      var _this = this;

      return h("input", {
        type: 'checkbox',
        attrs: this.v3 ? undefined : {
          type: 'checkbox',
          name: name,
          id: id || this.calculatedId,
          "aria-labelledby": ariaLabelledBy,
          "aria-describedby": ariaDescribedBy,
          disabled: disabled,
          tabIndex: kendo_vue_common_1.getTabIndex(tabIndex, disabled),
          role: 'checkbox',
          required: required !== undefined ? required : false,
          "aria-checked": this.computedChecked || this.checkedProp ? true : this.indeterminateProp ? 'mixed' : false,
          "aria-disabled": disabled || undefined
        },
        "class": inputClasses,
        name: name,
        id: id || this.calculatedId,
        ref: this.v3 ? function (el) {
          _this.inputRef = el;
        } : 'input',
        "aria-labelledby": ariaLabelledBy,
        "aria-describedby": ariaDescribedBy,
        checked: this.v3 ? Boolean(this.checkedProp) : null,
        domProps: this.v3 ? undefined : {
          "checked": Boolean(this.checkedProp)
        },
        disabled: disabled,
        tabIndex: kendo_vue_common_1.getTabIndex(tabIndex, disabled),
        role: 'checkbox',
        required: required !== undefined ? required : false,
        "aria-checked": this.computedChecked || this.checkedProp ? true : this.indeterminateProp ? 'mixed' : false,
        "aria-disabled": disabled || undefined,
        onChange: this.onChangeHandler,
        on: this.v3 ? undefined : {
          "change": this.onChangeHandler,
          "keydown": this.onKeyDownHandler,
          "focus": this.onFocusHandler,
          "blur": this.onBlurHandler
        },
        onKeydown: this.onKeyDownHandler,
        onFocus: this.onFocusHandler,
        onBlur: this.onBlurHandler
      });
    };

    if (labelRender) {
      var renderTemplate = labelRender ? kendo_vue_common_1.templateRendering.call(this, labelRender, kendo_vue_common_1.getListeners.call(this)) : null;
      renderedLabel = kendo_vue_common_1.getTemplate.call(this, {
        h: h,
        template: renderTemplate
      });
    }

    var checkboxLabel = function checkboxLabel() {
      return renderedLabel !== undefined ? h("label", {
        "class": 'k-checkbox-label',
        "for": id || this.calculatedId,
        attrs: this.v3 ? undefined : {
          "for": id || this.calculatedId
        },
        style: {
          userSelect: 'none'
        }
      }, [renderedLabel, labelOptional && h("span", {
        "class": "k-label-optional"
      }, [this.optionalMessage])]) : null;
    };

    return labelPlacement === 'before' ? h("span", {
      "class": checkboxClasses,
      dir: 'rtl',
      attrs: this.v3 ? undefined : {
        dir: 'rtl'
      }
    }, [checkboxInput.call(this), checkboxLabel.call(this), defaultSlot]) : h("span", {
      "class": checkboxClasses,
      dir: this.currentDir,
      attrs: this.v3 ? undefined : {
        dir: this.currentDir
      }
    }, [checkboxInput.call(this), checkboxLabel.call(this), defaultSlot]);
  },
  methods: {
    setValidity: function setValidity() {
      var isValid = this.$props.valid !== undefined ? this.$props.valid : !this.$props.required ? true : this.computedChecked ? true : false; // @ts-ignore

      if (this.input && this.input.setCustomValidity) {
        // @ts-ignore
        this.input.setCustomValidity(isValid ? '' : this.$props.validationMessage || this.defaultValidationMessage);
      }
    },
    localizeMessage: function localizeMessage(message) {
      return this.localizationService.toLanguageString(message, messages_1.messages[message]);
    },
    focusElement: function focusElement() {
      if (this.input) {
        // @ts-ignore
        this.input.focus();
      }
    },
    setValue: function setValue(e, val) {
      this.$data.valueDuringOnChange = val;
      var that = this;
      this.$nextTick(function () {
        if (!that.isCheckedControlled && !that.isValueControlled && !that.$props.disabled) {
          that.currentValue = val;
          that.currentChecked = val;
        }

        if (!that.$props.disabled) {
          var handle = {
            element: that.$el,
            focus: null // focusElement

          };
          that.$emit('change', {
            e: e,
            handle: handle,
            value: val
          });
          that.$emit('changemodel', val);
          that.$emit('update:modelValue', val);
        }

        this.$data.valueDuringOnChange = undefined;
      });
    },
    onChangeHandler: function onChangeHandler(e) {
      var newValue = e.target.checked;
      this.setValue(e, newValue);
    },
    onKeyDownHandler: function onKeyDownHandler(e) {
      if (this.$props.disabled) {
        return;
      }

      var keyCode = e.keyCode;
      var currentVal = e.currentTarget.checked;

      if (keyCode === kendo_vue_common_1.Keys.space) {
        e.preventDefault();
        e.stopPropagation();
        this.setValue(e, !currentVal);
      }
    },
    onBlurHandler: function onBlurHandler(e) {
      if (!this.$props.disabled) {
        this.$emit('blur', e);
      }
    },
    onFocusHandler: function onFocusHandler(e) {
      if (!this.$props.disabled) {
        this.$emit('focus', e);
      }
    }
  }
};
exports.Checkbox = Checkbox;
var CheckboxVue3 = Checkbox;
exports.CheckboxVue3 = CheckboxVue3;
