// @progress/kendo-vue-data-tools/dist/npm/filter/filters/BooleanFilter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BooleanFilterVue3 = exports.BooleanFilter = void 0; // @ts-ignore

var Vue = require("vue");

var allVue = Vue;
var gh = allVue.h;

var EnumFilter_1 = require("./EnumFilter");

var messages_1 = require("../../messages"); // tslint:enable:max-line-length


var BooleanFilter = {
  name: 'KendoBooleanFilter',
  props: {
    filter: {
      type: Object,
      required: true
    },
    dataItems: {
      type: Array,
      default: function _default() {
        return undefined;
      }
    },
    defaultItem: Object
  },
  data: function data() {
    return {
      currentData: [{
        text: messages_1.filterIsTrue,
        value: true
      }, {
        text: messages_1.filterIsFalse,
        value: false
      }]
    };
  },
  // @ts-ignore
  setup: !gh ? undefined : function () {
    var v3 = !!gh;
    return {
      v3: v3
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    return (// @ts-ignore
      h(EnumFilter_1.EnumFilter, {
        filter: this.$props.filter,
        attrs: this.v3 ? undefined : {
          filter: this.$props.filter,
          dataItems: this.$props.dataItems || this.currentData,
          defaultItem: this.$props.defaultItem
        },
        onFilterchange: this.handleFilterChange,
        on: this.v3 ? undefined : {
          "filterchange": this.handleFilterChange
        },
        dataItems: this.$props.dataItems || this.currentData,
        defaultItem: this.$props.defaultItem
      })
    );
  },
  methods: {
    handleFilterChange: function handleFilterChange(event) {
      this.$emit('filterchange', event);
    }
  }
};
exports.BooleanFilter = BooleanFilter;
var BooleanFilterVue3 = BooleanFilter;
exports.BooleanFilterVue3 = BooleanFilterVue3;
