// @progress/kendo-vue-data-tools/dist/npm/filter/filters/NumericFilter.js
"use strict";

var __assign = undefined && undefined.__assign || function () {
  __assign = Object.assign || function (t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
      s = arguments[i];

      for (var p in s) {
        if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
      }
    }

    return t;
  };

  return __assign.apply(this, arguments);
};

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NumericFilterVue3 = exports.NumericFilter = void 0; // @ts-ignore

var Vue = require("vue");

var allVue = Vue;
var gh = allVue.h;

var kendo_vue_inputs_1 = require("@progress/kendo-vue-inputs"); // tslint:enable:max-line-length


var NumericFilter = {
  name: 'kendoNumericFilter',
  props: {
    filter: {
      type: Object,
      required: true
    }
  },
  // @ts-ignore
  setup: !gh ? undefined : function () {
    var v3 = !!gh;
    return {
      v3: v3
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    var filter = this.$props.filter;
    return (// @ts-ignore
      h(kendo_vue_inputs_1.NumericTextBox, {
        value: typeof filter.value === 'number' ? filter.value : null,
        attrs: this.v3 ? undefined : {
          value: typeof filter.value === 'number' ? filter.value : null
        },
        onChange: this.onChange,
        on: this.v3 ? undefined : {
          "change": this.onChange
        }
      })
    );
  },
  methods: {
    onChange: function onChange(event) {
      this.$emit('filterchange', {
        nextFilter: __assign(__assign({}, this.$props.filter), {
          value: event.value
        })
      });
    }
  }
};
exports.NumericFilter = NumericFilter;
var NumericFilterVue3 = NumericFilter;
exports.NumericFilterVue3 = NumericFilterVue3;
