// @progress/kendo-vue-data-tools/dist/es/filter/Filter.js
var __assign = this && this.__assign || function () {
  __assign = Object.assign || function (t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
      s = arguments[i];

      for (var p in s) {
        if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
      }
    }

    return t;
  };

  return __assign.apply(this, arguments);
}; // @ts-ignore


import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
import { GroupFilter } from './GroupFilters';
import { GroupToolbar } from './GroupToolbar';
import { validatePackage, templateRendering, getListeners } from '@progress/kendo-vue-common';
import { packageMetadata } from '../package-metadata'; // tslint:enable:max-line-length

var Filter = {
  name: 'KendoFilter',
  // @ts-ignore
  emits: {
    change: null,
    'changemodel': null,
    'update:modelValue': null
  },
  model: {
    event: 'changemodel'
  },
  props: {
    fields: {
      type: Array,
      required: true
    },
    modelValue: {
      type: Object,
      default: undefined
    },
    value: Object,
    defaultGroupFilter: Object
  },
  created: function created() {
    validatePackage(packageMetadata);
  },
  // @ts-ignore
  setup: !gh ? undefined : function () {
    var v3 = !!gh;
    return {
      v3: v3
    };
  },
  computed: {
    computedValue: function computedValue() {
      var value;

      if (this.$props.value !== undefined) {
        value = this.$props.value;
      } else if (this.$props.modelValue !== undefined) {
        value = this.$props.modelValue;
      }

      return value;
    }
  },
  render: function render(createElement) {
    var h = gh || createElement;
    var fields = this.$props.fields.map(function (field) {
      return __assign(__assign({}, field), {
        filterRender: templateRendering.call(this, field.filterRender, getListeners.call(this))
      });
    }, this);
    return h("div", {
      "class": 'k-widget k-filter'
    }, [h("ul", {
      "class": "k-filter-container"
    }, [h("li", {
      "class": "k-filter-group-main"
    }, [// @ts-ignore function children
    h(GroupToolbar, {
      "class": "k-filter-group-main",
      filter: this.computedValue,
      attrs: this.v3 ? undefined : {
        filter: this.computedValue,
        fields: fields,
        defaultGroupFilter: this.$props.defaultGroupFilter || {
          logic: 'and',
          filters: []
        }
      },
      fields: fields,
      onChange: this.onFilterChange,
      on: this.v3 ? undefined : {
        "change": this.onFilterChange,
        "remove": this.onGroupRemove
      },
      onRemove: this.onGroupRemove,
      defaultGroupFilter: this.$props.defaultGroupFilter || {
        logic: 'and',
        filters: []
      }
    }), // @ts-ignore function children
    h(GroupFilter, {
      "class": "k-filter-group-main",
      filter: this.computedValue,
      attrs: this.v3 ? undefined : {
        filter: this.computedValue,
        fields: fields,
        defaultGroupFilter: this.$props.defaultGroupFilter || {
          logic: 'and',
          filters: []
        }
      },
      fields: fields,
      onChange: this.onFilterChange,
      on: this.v3 ? undefined : {
        "change": this.onFilterChange,
        "remove": this.onGroupRemove
      },
      onRemove: this.onGroupRemove,
      defaultGroupFilter: this.$props.defaultGroupFilter || {
        logic: 'and',
        filters: []
      }
    })])])]);
  },
  methods: {
    onFilterChange: function onFilterChange(event) {
      var changeEvent = {
        filter: event.nextFilter,
        event: event.event,
        target: this
      };
      this.$emit('change', changeEvent);
      this.$emit('changemodel', event.nextFilter);
      this.$emit('update:modelValue', event.nextFilter);
    },
    onGroupRemove: function onGroupRemove(event) {
      var nextFilter = __assign(__assign({}, this.computedValue), {
        filters: []
      });

      var changeEvent = {
        filter: nextFilter,
        event: event.event,
        target: this
      };
      this.$emit('change', changeEvent);
      this.$emit('changemodel', nextFilter);
      this.$emit('update:modelValue', nextFilter);
    }
  }
};
var FilterVue3 = Filter;
export { Filter, FilterVue3 };
