// @progress/kendo-vue-labels/dist/npm/Hint.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HintVue3 = exports.Hint = void 0; // @ts-ignore

var Vue = require("vue");

var allVue = Vue;
var gh = allVue.h;

var package_metadata_1 = require("./package-metadata");

var kendo_vue_common_1 = require("@progress/kendo-vue-common");
/**
 * Represents the default `Hint` component.
 */


var Hint = {
  name: 'KendoHint',
  props: {
    id: String,
    direction: {
      type: String,
      default: 'start',
      validator: function validator(value) {
        return ['start', 'end'].includes(value);
      }
    }
  },
  created: function created() {
    kendo_vue_common_1.validatePackage(package_metadata_1.packageMetadata);
  },
  computed: {
    hintClassName: function hintClassName() {
      return {
        'k-form-hint': true,
        'k-text-start': this.$props.direction === 'start',
        'k-text-end': this.$props.direction === 'end'
      };
    }
  },
  // @ts-ignore
  setup: !gh ? undefined : function () {
    var v3 = !!gh;
    return {
      v3: v3
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    var defaultSlots = kendo_vue_common_1.getDefaultSlots(this);
    return h("div", {
      id: this.$props.id,
      attrs: this.v3 ? undefined : {
        id: this.$props.id
      },
      "class": this.hintClassName
    }, [defaultSlots]);
  }
};
exports.Hint = Hint;
var HintVue3 = Hint;
exports.HintVue3 = HintVue3;
