// @progress/kendo-vue-labels/dist/npm/Label.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LabelVue3 = exports.Label = void 0; // @ts-ignore

var Vue = require("vue");

var allVue = Vue;
var gh = allVue.h;
var inject = allVue.inject;

var kendo_vue_common_1 = require("@progress/kendo-vue-common");

var messages_1 = require("./messages");

var kendo_vue_intl_1 = require("@progress/kendo-vue-intl");

var package_metadata_1 = require("./package-metadata");
/**
 * Represents the default `Label` component.
 */


var Label = {
  name: 'KendoLabel',
  props: {
    id: String,
    editorId: String,
    editorRef: String,
    editorValid: {
      type: Boolean,
      default: undefined
    },
    editorDisabled: Boolean,
    optional: Boolean
  },
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  created: function created() {
    kendo_vue_common_1.validatePackage(package_metadata_1.packageMetadata);
  },
  computed: {
    labelClassName: function labelClassName() {
      return {
        'k-label': true,
        'k-label-empty': !this.defaultSlots,
        'k-text-error': this.$props.editorValid === false,
        'k-text-disabled': this.$props.editorDisabled === true
      };
    }
  },
  methods: {
    onLabelClick: function onLabelClick(e) {
      if (this.$props.editorRef && this.$props.editorRef.current && !this.$props.editorDisabled) {
        if (this.$props.editorRef.current.focus) {
          e.preventDefault();
          this.$props.editorRef.current.focus();
        }

        var editorActionElement = this.$props.editorRef.current.actionElement;

        if (editorActionElement) {
          e.preventDefault();
          editorActionElement.click();
        }
      }
    }
  },
  // @ts-ignore
  setup: !gh ? undefined : function () {
    var v3 = !!gh;
    var kendoLocalizationService = inject('kendoLocalizationService', {});
    return {
      v3: v3,
      kendoLocalizationService: kendoLocalizationService
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    this.defaultSlots = kendo_vue_common_1.getDefaultSlots(this);
    var ls = kendo_vue_intl_1.provideLocalizationService(this);
    var _a = this.$props,
        id = _a.id,
        editorId = _a.editorId,
        optional = _a.optional;
    var localizedOptional = optional ? ls.toLanguageString(messages_1.labelsOptional, messages_1.messages[messages_1.labelsOptional]) : '';
    var optionalElement = localizedOptional && h("span", {
      "class": 'k-label-optional'
    }, [localizedOptional]);
    return h("label", {
      id: id,
      attrs: this.v3 ? undefined : {
        id: id,
        "for": editorId
      },
      "for": editorId,
      onClick: this.onLabelClick,
      on: this.v3 ? undefined : {
        "click": this.onLabelClick
      },
      "class": this.labelClassName
    }, [this.defaultSlots, optionalElement]);
  }
};
exports.Label = Label;
var LabelVue3 = Label;
exports.LabelVue3 = LabelVue3;
