// @progress/kendo-vue-labels/dist/npm/main.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HintVue3 = exports.Hint = exports.ErrorVue3 = exports.Error = exports.LabelVue3 = exports.Label = exports.FloatingLabelVue3 = exports.FloatingLabel = void 0;
var Label_1 = require("./Label");
Object.defineProperty(exports, "Label", { enumerable: true, get: function () { return Label_1.Label; } });
Object.defineProperty(exports, "LabelVue3", { enumerable: true, get: function () { return Label_1.LabelVue3; } });
var Error_1 = require("./Error");
Object.defineProperty(exports, "Error", { enumerable: true, get: function () { return Error_1.Error; } });
Object.defineProperty(exports, "ErrorVue3", { enumerable: true, get: function () { return Error_1.ErrorVue3; } });
var Hint_1 = require("./Hint");
Object.defineProperty(exports, "Hint", { enumerable: true, get: function () { return Hint_1.Hint; } });
Object.defineProperty(exports, "HintVue3", { enumerable: true, get: function () { return Hint_1.HintVue3; } });
var FloatingLabel_1 = require("./FloatingLabel");
Object.defineProperty(exports, "FloatingLabel", { enumerable: true, get: function () { return FloatingLabel_1.FloatingLabel; } });
Object.defineProperty(exports, "FloatingLabelVue3", { enumerable: true, get: function () { return FloatingLabel_1.FloatingLabelVue3; } });
