// @progress/kendo-charts-vue-wrapper/dist/es/KendoSparkline/KendoSparklineMixin.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _kendoBaseComponentsVueWrapper = require('@progress/kendo-base-components-vue-wrapper');

var _SparklineProps = require('./SparklineProps');

var _SparklineProps2 = _interopRequireDefault(_SparklineProps);

var _KendoSparklineSeriesItem = require('./KendoSparklineSeriesItem');

var _KendoSparklineSeriesItem2 = _interopRequireDefault(_KendoSparklineSeriesItem);

var _CategoryAxis = require('./CategoryAxis');

var _CategoryAxis2 = _interopRequireDefault(_CategoryAxis);

var _ChartArea = require('./ChartArea');

var _ChartArea2 = _interopRequireDefault(_ChartArea);

var _PlotArea = require('./PlotArea');

var _PlotArea2 = _interopRequireDefault(_PlotArea);

var _SeriesDefaults = require('./SeriesDefaults');

var _SeriesDefaults2 = _interopRequireDefault(_SeriesDefaults);

var _Tooltip = require('./Tooltip');

var _Tooltip2 = _interopRequireDefault(_Tooltip);

var _ValueAxis = require('./ValueAxis');

var _ValueAxis2 = _interopRequireDefault(_ValueAxis);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
    props: _SparklineProps2.default,
    emits: {
        'kendowidgetready': null
    },
    mixins: [_kendoBaseComponentsVueWrapper.KendoBaseComponent, _kendoBaseComponentsVueWrapper.KendoBaseDatasourceComponent, _CategoryAxis2.default, _ChartArea2.default, _PlotArea2.default, _SeriesDefaults2.default, _Tooltip2.default, _ValueAxis2.default],
    mounted: function mounted() {
        this._resolveChildren();

        kendo.jQuery(this.$el).kendoSparkline(this.widgetOptions);

        if (this.v3) {
            this.ready();
            this.$emit('kendowidgetready', this.kendoWidget());
        } else {
            this.$emit('kendowidgetready', this.kendoWidget());
        }
    },

    methods: {
        kendoWidget: function kendoWidget() {
            return kendo.jQuery(this.$el).getKendoSparkline();
        },
        _resolveChildren: function _resolveChildren() {
            this.resolveChildren('series', _KendoSparklineSeriesItem2.default.name);
        }
    }
};
