// @progress/kendo-barcodes-vue-wrapper/dist/npm/KendoBarcode/KendoBarcodeMixin.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _kendoBaseComponentsVueWrapper = require('@progress/kendo-base-components-vue-wrapper');

var _BarcodeProps = require('./BarcodeProps');

var _BarcodeProps2 = _interopRequireDefault(_BarcodeProps);

var _Border = require('./Border');

var _Border2 = _interopRequireDefault(_Border);

var _Padding = require('./Padding');

var _Padding2 = _interopRequireDefault(_Padding);

var _Text = require('./Text');

var _Text2 = _interopRequireDefault(_Text);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
    props: _BarcodeProps2.default,
    emits: {
        'kendowidgetready': null
    },
    mixins: [_kendoBaseComponentsVueWrapper.KendoBaseComponent, _Border2.default, _Padding2.default, _Text2.default],
    mounted() {
        kendo.jQuery(this.$el).kendoBarcode(this.widgetOptions);

        if (this.v3) {
            this.ready();
            this.$emit('kendowidgetready', this.kendoWidget());
        } else {
            this.$emit('kendowidgetready', this.kendoWidget());
        }
    },
    methods: {
        kendoWidget() {
            return kendo.jQuery(this.$el).getKendoBarcode();
        }
    }
};
