// @progress/kendo-vue-buttons/dist/es/ListButton/ButtonItem.js
// @ts-ignore
import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
import { getTemplate } from '@progress/kendo-vue-common'; // tslint:enable:max-line-length

var ButtonItem = {
  name: 'KendoButtonItem',
  // @ts-ignore
  emits: {
    click: null,
    down: null
  },
  props: {
    focused: Boolean,
    index: Number,
    item: Object,
    render: [String, Object, Function],
    dataItem: [String, Object],
    id: String,
    textField: String
  },
  computed: {
    wrapperClass: function wrapperClass() {
      var _a = this.$props,
          dataItem = _a.dataItem,
          focused = _a.focused;
      return {
        'k-item': true,
        'k-state-focused': focused,
        'k-state-selected': dataItem.selected,
        'k-state-disabled': dataItem.disabled
      };
    }
  },
  methods: {
    onClick: function onClick(event) {
      this.$emit('click', event, this.$props.index);
    },
    onDown: function onDown(event) {
      this.$emit('down', event, this.$props.index);
    }
  },
  // @ts-ignore
  setup: !gh ? undefined : function () {
    var v3 = !!gh;
    return {
      v3: v3
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    var _a = this.$props,
        dataItem = _a.dataItem,
        id = _a.id,
        render = _a.render;

    var renderContent = function renderContent() {
      var _a = this.$props,
          textField = _a.textField,
          index = _a.index;
      var text = dataItem.text !== undefined ? dataItem.text : textField ? dataItem[textField] : dataItem;
      var iconClass = dataItem.icon ? "k-icon k-i-" + dataItem.icon : dataItem.iconClass;
      var itemContent = [iconClass && h("span", {
        "class": iconClass,
        role: "presentation",
        attrs: this.v3 ? undefined : {
          role: "presentation"
        },
        key: "icon"
      }), dataItem.imageUrl && h("img", {
        "class": "k-image",
        alt: "",
        attrs: this.v3 ? undefined : {
          alt: "",
          src: dataItem.imageUrl,
          role: "presentation"
        },
        src: dataItem.imageUrl,
        role: "presentation",
        key: "image"
      }), text];
      return getTemplate.call(this, {
        h: h,
        template: this.$props.dataItem.render || render,
        defaultRendering: itemContent,
        additionalProps: {
          item: dataItem,
          itemIndex: index
        }
      });
    };

    var item = h("li", {
      id: id,
      attrs: this.v3 ? undefined : {
        id: id,
        role: "menuItem",
        "aria-disabled": dataItem.disabled || undefined
      },
      "class": this.wrapperClass,
      onClick: this.onClick,
      on: this.v3 ? undefined : {
        "click": this.onClick,
        "mousedown": this.onDown,
        "pointerdown": this.onDown
      },
      onMousedown: this.onDown,
      onPointerdown: this.onDown,
      role: "menuItem",
      "aria-disabled": dataItem.disabled || undefined
    }, [renderContent.call(this)]);
    return item;
  }
};
var ButtonItemVue3 = ButtonItem;
export { ButtonItem, ButtonItemVue3 };
