// @progress/kendo-vue-labels/dist/es/Label.js
// @ts-ignore
import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
var inject = allVue.inject;
import { getDefaultSlots, validatePackage } from '@progress/kendo-vue-common';
import { labelsOptional, messages } from './messages';
import { provideLocalizationService } from '@progress/kendo-vue-intl';
import { packageMetadata } from './package-metadata';
/**
 * Represents the default `Label` component.
 */

var Label = {
  name: 'KendoLabel',
  props: {
    id: String,
    editorId: String,
    editorRef: String,
    editorValid: {
      type: Boolean,
      default: undefined
    },
    editorDisabled: Boolean,
    optional: Boolean
  },
  inject: {
    kendoLocalizationService: {
      default: null
    }
  },
  created: function created() {
    validatePackage(packageMetadata);
  },
  computed: {
    labelClassName: function labelClassName() {
      return {
        'k-label': true,
        'k-label-empty': !this.defaultSlots,
        'k-text-error': this.$props.editorValid === false,
        'k-text-disabled': this.$props.editorDisabled === true
      };
    }
  },
  methods: {
    onLabelClick: function onLabelClick(e) {
      if (this.$props.editorRef && this.$props.editorRef.current && !this.$props.editorDisabled) {
        if (this.$props.editorRef.current.focus) {
          e.preventDefault();
          this.$props.editorRef.current.focus();
        }

        var editorActionElement = this.$props.editorRef.current.actionElement;

        if (editorActionElement) {
          e.preventDefault();
          editorActionElement.click();
        }
      }
    }
  },
  // @ts-ignore
  setup: !gh ? undefined : function () {
    var v3 = !!gh;
    var kendoLocalizationService = inject('kendoLocalizationService', {});
    return {
      v3: v3,
      kendoLocalizationService: kendoLocalizationService
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    this.defaultSlots = getDefaultSlots(this);
    var ls = provideLocalizationService(this);
    var _a = this.$props,
        id = _a.id,
        editorId = _a.editorId,
        optional = _a.optional;
    var localizedOptional = optional ? ls.toLanguageString(labelsOptional, messages[labelsOptional]) : '';
    var optionalElement = localizedOptional && h("span", {
      "class": 'k-label-optional'
    }, [localizedOptional]);
    return h("label", {
      id: id,
      attrs: this.v3 ? undefined : {
        id: id,
        "for": editorId
      },
      "for": editorId,
      onClick: this.onLabelClick,
      on: this.v3 ? undefined : {
        "click": this.onLabelClick
      },
      "class": this.labelClassName
    }, [this.defaultSlots, optionalElement]);
  }
};
var LabelVue3 = Label;
export { Label, LabelVue3 };
